### PR TITLE
Added onboarding form

### PR DESCRIPTION
**Pull request by ayanb12**

### CODE REVIEW

**Comment by robinsingh1912** [comment]
@ayanb12  can you please resolve the above conflicts first

**Comment by ayanb12** [comment]
> 

Yes working on it

**Review by theisens — APPROVED** [review]

File: src/components/Checkbox.jsx [hunk 1]
```diff
@@ -18,6 +18,9 @@ export default function CheckBox({ label, id }) {
           onClick={handleCheckboxChange}
           checked={isChecked}
           className='accent-white text-black h-7 w-7 rounded-none border-none'
+          {...register(id, {
```
Comment (robinsingh1912) [inline src/components/Checkbox.jsx:21]: Please add and Use zod for input validation

File: src/components/OnboardingForm.jsx [hunk 2]
```diff
@@ -1,63 +1,91 @@
...
-        Let&apos;s go
-      </button>
-    </form>
+          <div className='w-full relative flex flex-col min-w-[300p[x] max-w-[400px] text-left font-futBook'>
+            <input
+              id='nickname'
+              type='text'
+              placeholder='Nickname'
+              className='peer h-16 w-full shadow-none outline-none pl-6 text-base text-black placeholder-transparent'
+              {...register('nickname', {
```
Comment (robinsingh1912) [inline src/components/OnboardingForm.jsx:38]: Zod resolver here too with react hook form

**Review by robinsingh1912 — CHANGES_REQUESTED** [review]

File: src/components/OnboardingForm.jsx (hunk 2 above)
Comment (ayanb12) [reply src/components/OnboardingForm.jsx]: Added

File: src/components/Checkbox.jsx (hunk 1 above)
Comment (ayanb12) [reply src/components/Checkbox.jsx]: Added

**Review by robinsingh1912 — CHANGES_REQUESTED** [review]

File: src/components/NavBar.jsx [hunk 3]
```diff
@@ -9,10 +9,23 @@ import DropdownUserMenu from './DropdownUserMenu';
...
+        setShowModal(true);
+        setSignInStep(2);
+      }
+    }
+  };
+
+  useEffect(() => {
+    openOnboardingForm(isAuthenticated, user);
+  }, [isAuthenticated, user]);
+
```
[the remark this answers is not in the document]
Comment (ayanb12) [reply src/components/NavBar.jsx:28]: Removed the nested parameters

File: src/components/NavBar.jsx [hunk 4]
```diff
@@ -26,15 +39,13 @@ export default function Navbar() {
...
-          handleCreateAccount={() => {
-            setShowModal(false);
-          }}
           handleEmailSubmit={handleEmailSubmit}
           handleCloseEmail={() => {
             setSignInStep(0);
             handleCloseModal();
           }}
           handleClose={handleCloseModal}
+          logout={logout}
```
[the remark this answers is not in the document]
Comment (ayanb12) [reply src/components/NavBar.jsx:48]: corrected used from useAuth

File: src/components/DropdownInput.jsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ const countries = [
   { id: 'MEX', name: 'Mexico' },
 ];
 
-export default function DropdownInput({ register, setValue }) {
+export default function DropdownInput(props) {
```
Comment (robinsingh1912) [inline src/components/DropdownInput.jsx:17]: should destructure the props

File: src/hooks/useValidations.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    if (values.includes(curr)) return { ...acc, [curr]: schema[curr] };
+    else return acc;
+  }, {});
+
+  return {
+    formSchema: z.object(filteredSchema),
+  };
+};
+
+export default useValidations;
```
Comment (robinsingh1912) [inline src/hooks/useValidations.js:29]: No need to create a hook If your helper function does not require any React hooks, then you do not need to create a custom hook. A simple utility function or helper function would sufficient.

**Review by robinsingh1912 — CHANGES_REQUESTED** [review]

File: src/components/Forms/Controller/Controller.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      props.onChange({
+        target: {
+          name,
+          value: (e.target && e.target.value) || val,
+        },
+      }),
+    onBlur: props.onBlur,
+    name: props.name,
+  });
+};
```
Comment (robinsingh1912) [inline src/components/Forms/Controller/Controller.jsx:15]: import { useForm, Controller } from "react-hook-form";
 and i guess no need to create create a component we can directly import it

Comment (ayanb12) [reply src/components/Forms/Controller/Controller.jsx:15]: Okay Got it

File: src/hooks/useValidations.js (hunk 6 above)
Comment (ayanb12) [reply src/hooks/useValidations.js]: Okay moving the logic to utils

**Review by robinsingh1912 — APPROVED** [review]

**Pull request merged**